### PR TITLE
Replace the bartenders Riot shotgun with a Sawn off break action shotgun

### PR DESCRIPTION
**Pull request by Sir-Mora**
## About the PR 
replaces the bartenders riot shotgun with a sawn off shotgun, pretty self explanatory.
thanks to @cogwerks for the sawn-offs sprite

The Sawn off's details are
- 2 round capacity
- normal sized item
- fits in one hand due to being sawn-off
- can only be reloaded when open
- can only be shot when closed
- fires one shot per trigger pull

## Why's this needed? 
I think this should be added because A: sawn off shotguns are cool, and B: the bartender having a item that is other wise limited to the armory is strange, so changing that makes the bartender make more sense

## Changelog

```changelog
(u)Mora
(*) Replaced the bartenders shotgun with a 2-shot break action sawn off shotgun, added extra box of rubber slugs
```

### CODE REVIEW

**Comment by Zonespace27** [comment]
> ## Changelog
> ```
> (u)Mora
> (*)Added a Break action sawn off shotgun to the game - the details of which are under the about the pr section
> (*)Gave the bartender another mag of riot rounds - to compensate for lost ammo
> (+) changed the book "bartending pocket guide" - riotgun was replaced with sawn off
> (+) added a new ammo type - prevents issues with the gun spawning with 8 shots when it should have had 2
> ```
This is the changelog that goes in game, so I'd suggest getting rid of mention of details in the PR section

**Comment by Sir-Mora** [comment]
> > ## Changelog
> > ```
> > (u)Mora
> > (*)Added a Break action sawn off shotgun to the game - the details of which are under the about the pr section
> > (*)Gave the bartender another mag of riot rounds - to compensate for lost ammo
> > (+) changed the book "bartending pocket guide" - riotgun was replaced with sawn off
> > (+) added a new ammo type - prevents issues with the gun spawning with 8 shots when it should have had 2
> > ```
> 
> This is the changelog that goes in game, so I'd suggest getting rid of mention of details in the PR section

fixed it by adding the details to the changelog, not sure if thats the way thats supposed to be done but i did it that way

**Comment by Yellow-Mushroom** [comment]
Can you add a pic of the shotgun?

**Comment by flappybatpal** [comment]
Are there some typos in the alternate names?
Street Hoitzer
Twin 12 Guage
Dr. Bullous' Tunder-clapper

Probably worth using a \ to shorten the horizontal length of the names line. Also some inconsistency in them being capitalized.

**Comment by BatElite** [comment]
Shitty crop of icon diff bot for the sprites:
afbeelding

Is the bartender going to get the 6 other bullets the riot shotgun used to have somewhere? Only having 2 at all would probably mean they only ever get used for antags.
(Not counting hacking the vendor for bullets, that seems like needn't be standard bartender stuff)

**Comment by Sir-Mora** [comment]
> Shitty crop of icon diff bot for the sprites:
> afbeelding
> 
> Is the bartender going to get the 6 other bullets the riot shotgun used to have somewhere? Only having 2 at all would probably mean they only ever get used for antags.
> (Not counting hacking the vendor for bullets, that seems like needn't be standard bartender stuff)

yes, the bartender will be starting with a extra full mag of riot rounds in their inventory to compensate for the lost ammo.

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
First off, I want to say that this is really cool and something I've personally wanted to see for awhile, more firearms with cool interactions beyond just "shoot" are neat and make gunplay more interesting. The reloading and function of this shotgun feels really good, how it only ejects the empty shells when you open the action, and how you actually need to open said action to insert new shells. I like the sprites too.

There are a few issues: 
- the right hand inhands are incorrect when you're facing east and west, they're using the left hand inhands layout, the hands need to be swapped
- the section in the bartending guide has multiple grammatical and spelling errors
- many of the names that can be chosen for the gun have inconsistent capitalization or spelling errors
- the status messages for reloading do not grammatically work for many of the names (needs a "the" before the name, missing punctuation)
- personally I think that the status messages should be a little more clear in regards to the intent, ex. `You cant reach through the Double Barreled Shotgun to take shells out! You'll have to open Double Barreled Shotgun first!` could really just say `The Double Barreled Shotgun is still closed, you need to open the action to take the shells out!`

Lastly (this is just a personal thing, my opinion), it should have its own firing sound effect and possibly some unique opening/closing audio, but that's not a requirement and it's something I can add later. I can't speak to the code itself so I'll let others review that, but functionally it works and is fun to use!

**Comment by Sir-Mora** [comment]
> First off, I want to say that this is really cool and something I've personally wanted to see for awhile, more firearms with cool interactions beyond just "shoot" are neat and make gunplay more interesting. The reloading and function of this shotgun feels really good, how it only ejects the empty shells when you open the action, and how you actually need to open said action to insert new shells. I like the sprites too.
> 
> There are a few issues:
> 
> * the right hand inhands are incorrect when you're facing east and west, they're using the left hand inhands layout, the hands need to be swapped
> * the section in the bartending guide has multiple grammatical and spelling errors
> * many of the names that can be chosen for the gun have inconsistent capitalization or spelling errors
> * the status messages for reloading do not grammatically work for many of the names (needs a "the" before the name, missing punctuation)
> * personally I think that the status messages should be a little more clear in regards to the intent, ex. `You cant reach through the Double Barreled Shotgun to take shells out! You'll have to open Double Barreled Shotgun first!` could really just say `The Double Barreled Shotgun is still closed, you need to open the action to take the shells out!`
> 
> Lastly (this is just a personal thing, my opinion), it should have its own firing sound effect and possibly some unique opening/closing audio, but that's not a requirement and it's something I can add later. I can't speak to the code itself so I'll let others review that, but functionally it works and is fun to use!

all of the above points are true, they should be corrected soon.

> First off, I want to say that this is really cool and something I've personally wanted to see for awhile, more firearms with cool interactions beyond just "shoot" are neat and make gunplay more interesting. The reloading and function of this shotgun feels really good, how it only ejects the empty shells when you open the action, and how you actually need to open said action to insert new shells. I like the sprites too.
> 
> There are a few issues:
> 
> * the right hand inhands are incorrect when you're facing east and west, they're using the left hand inhands layout, the hands need to be swapped
> * the section in the bartending guide has multiple grammatical and spelling errors
> * many of the names that can be chosen for the gun have inconsistent capitalization or spelling errors
> * the status messages for reloading do not grammatically work for many of the names (needs a "the" before the name, missing punctuation)
> * personally I think that the status messages should be a little more clear in regards to the intent, ex. `You cant reach through the Double Barreled Shotgun to take shells out! You'll have to open Double Barreled Shotgun first!` could really just say `The Double Barreled Shotgun is still closed, you need to open the action to take the shells out!`
> 
> Lastly (this is just a personal thing, my opinion), it should have its own firing sound effect and possibly some unique opening/closing audio, but that's not a requirement and it's something I can add later. I can't speak to the code itself so I'll let others review that, but functionally it works and is fun to use!

thanks for the feedback! I somehow didn't notice these problems and have (hopefully) fixed them 👍

**Comment by Tarmunora** [comment]
way too much changelog

**Review by virvatuli-gh — COMMENTED** [review]
Was looking at the sprite itself and noticed that there are actually some noticeable differences between the closed and open states in regards to shading at the rear of the receiver and grip. From what I remember Cogwerks' final versions of the sprite didn't have these differences, you might wanna talk to them to get it sorted!
closed
open
Other than that and the minor things I noted, the changelog only really needs to be 1 line. People can click on the changelog to be brought right to the PR on github here so they can see all the changes listed in the PR description instead! I don't know if you talked to Cogwerks already about it, but you should ask how/if they want to be attributed in the log. Example:
`(*) Replaced the bartenders shotgun with a break action sawn off shotgun, sprites by Cogwerks`

File: code/obj/item/gun/kinetic.dm [hunk 1]
```diff
@@ -1743,3 +1743,71 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
...
+			src.casings_to_eject = src.shells_to_eject
+
+			if (src.casings_to_eject > 0)
+				src.ejectcasings()
+				src.shells_to_eject = 0
+		..()
+
+	attackby(obj/item/b as obj, mob/user as mob)
+		if (istype(b, /obj/item/ammo/bullets) && src.broke_open == FALSE)
+			boutput(user, "<span class='alert'>You can't shove shells down the barrel! You'll have to open the [src] first!</span>")
```
[the remark this answers is not in the document]
Comment (Sir-Mora) [reply code/obj/item/gun/kinetic.dm]: that makes more sense, its been fixed

File: code/obj/item/gun/kinetic.dm [hunk 2]
```diff
@@ -1743,3 +1743,71 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
...
+	two_handed = 0
+	has_empty_state = 0
+	gildable = 0
+	add_residue = 1
+	var/broke_open = FALSE
+	var/shells_to_eject = 0
+
+	New() //uses a special box of ammo that only starts with 2 shells to prevent issues with overloading
+		if(prob(25))
+			name = pick ("Bessie", "Mule", "Loud Louis", "Boomstick", "Coach Gun", "Shorty", "Sawn-off Shotgun", "Street Sweeper", "Street Howitzer", "Big Boy", "Slugger", "Closing Time", "Garbage Day", "Rooty Tooty Point and Shooty", "Twin 12 Gauge", "Master Blaster", "Ass Blaster", "Blunderbuss", "Dr. Bullous' Tunder-Clapper", "Super Shotgun", "Insurance Policy", "Last Call", "Super-Duper Shotgun")
```
[the remark this answers is not in the document]
Comment (Sir-Mora) [reply code/obj/item/gun/kinetic.dm]: that wasn't intentional, it's been fixed

**Comment by Sir-Mora** [comment]
> Was looking at the sprite itself and noticed that there are actually some noticeable differences between the closed and open states in regards to shading at the rear of the receiver and grip. From what I remember Cogwerks' final versions of the sprite didn't have these differences, you might wanna talk to them to get it sorted!
> closed
> open
> Other than that and the minor things I noted, the changelog only really needs to be 1 line. People can click on the changelog to be brought right to the PR on github here so they can see all the changes listed in the PR description instead! I don't know if you talked to Cogwerks already about it, but you should ask how/if they want to be attributed in the log. Example:
> `(*) Replaced the bartenders shotgun with a break action sawn off shotgun, sprites by Cogwerks`

the sprite thing was caused by me having to shorten the sprite to get it to fit in one hand, so thats been fixed. I also updated the changlog to be shorter.

**Comment by Tarmunora** [comment]
trimmed changelog

**Comment by virvatuli-gh** [comment]
You have multiple merge conflicts that you need to resolve, I'll remove the Stale label so this doesn't get autoclosed but in order for it to be reviewed again and possibly merged these conflicts need to be fixed first!

`code/datums/jobs.dm`
`icons/mob/inhand/hand_weapons.dmi`
`icons/obj/items/gun.dmi`

**Comment by Sir-Mora** [comment]
gonna close this due to lack of interest and merge conflicts. may come back to it later and redo it

**Pull request closed without merging**